lib/src/file_util.rs

```diff
@@ -42,6 +42,22 @@ impl<T> IoResultExt<T> for io::Result<T> {
     }
 }
 
+/// Creates a directory or does nothing if the directory already exists.
+///
+/// Returns the &Path if successful, so it can be chained with .and_then() and
+/// friends which take a closure with a single param.
+///
+/// Returns the underlying error if the directory can't be created.
+/// The function will also fail if intermediate directories on the path do not
+/// already exist.
+pub fn create_or_reuse_dir(dirname: &Path) -> io::Result<&Path> {
```

Comment:
nit: `-> io::Result<()>`? The return value no longer makes sense.

Reply:
Good point. Done.

Thanks!